packages/hooks/src/use-data-table-sorting-state/use-data-table-sorting-state.js

```diff
@@ -0,0 +1,25 @@
+import { useState, useCallback } from 'react';
+
+const defaultValues = { key: 'createdAt', order: 'desc' };
+const applyIf = (values, key) => (values[key] ? { key } : {});
+
+const useDataTableSortingState = (initialValues) => {
+  const mergedValues = {
+    ...defaultValues,
+    ...applyIf(initialValues, 'key'),
+    ...applyIf(initialValues, 'order'),
```

Comment:
I preferred to take this approach over `pick`.
My opinion was that if the consumer misses to provide a key or order, the hook is "resilient" to catch and fallback to the defined default values.

![image](https://user-images.githubusercontent.com/462507/113151023-fd445280-9234-11eb-8115-08526856e5ea.png)